.github/workflows/update-charts.yml

```diff
@@ -0,0 +1,109 @@
+name: Update helm charts
+
+on:
+  repository_dispatch:
+    types: [update-chart]
+
+jobs:
+  check-update-type:
+    name: Detect update type
+    runs-on: ubuntu-latest
+    outputs:
+      update_type: ${{ steps.check_update_type.outputs.update_type }}
+    steps:
+      - name: Install semver comparison tool
+        run: |
+          wget -O ./semver https://github.com/fsaintjacques/semver-tool/raw/3.4.0/src/semver
+          chmod +x ./semver
```

Comment:
Nitpicking, you can download the binary to a dedicated dir and then "register" this dir into `$GITHUB_PATH` to make the command available (and save the `./semver` thing when invoking it).

Something like we're doing here:
https://github.com/kubewarden/github-actions/blob/18f8be2e8d95bfbb72333cad1bebbe48c9b06827/kwctl-installer/action.yaml#L16-L29